src/StackExchange.Metrics/MetricsCollector.cs

```diff
@@ -139,32 +93,36 @@ private class RootMetricInfo
         public MetricsCollector(MetricsCollectorOptions options)
         {
             ExceptionHandler = options.ExceptionHandler ?? (_ => { });
-            MetricsNamePrefix = options.MetricsNamePrefix ?? "";
-            if (MetricsNamePrefix != "" && !MetricValidation.IsValidMetricName(MetricsNamePrefix))
-                throw new Exception("\"" + MetricsNamePrefix + "\" is not a valid metric name prefix.");
-
-            _endpoints = options.Endpoints?.ToImmutableArray() ?? ImmutableArray<MetricEndpoint>.Empty;
-            _sets = options.Sets?.ToImmutableArray() ?? ImmutableArray<IMetricSet>.Empty;
-
             ThrowOnPostFail = options.ThrowOnPostFail;
             ThrowOnQueueFull = options.ThrowOnQueueFull;
             ReportingInterval = options.SnapshotInterval;
             FlushInterval = options.FlushInterval;
             RetryInterval = options.RetryInterval;
-            PropertyToTagName = options.PropertyToTagName;
-            TagValueConverter = options.TagValueConverter;
-            DefaultTags = ValidateDefaultTags(options.DefaultTags);
+            RetryCount = options.RetryCount;
```

Comment:
Maybe it's be better to *not* have all these props, and hold on to the options ref instead? That gets us closer to an `IOptions<T>.Value` approach for later, allowing changing props on the fly and such.

Reply:
Possibly, but there's a whole bunch of complication around managing endpoints and sources which are effectively frozen at time of the collector being instantiated. +1 for the simple props though